src/cpp/fastdds/subscriber/DataReaderImpl.cpp

```diff
@@ -1376,8 +1376,6 @@ void DataReaderImpl::release_payload_pool()
 
     PoolConfig config = PoolConfig::from_history_attributes(history_.m_att);
     payload_pool_->release_history(config, true);
-
```

Comment:
Shouldn't we do `payload_pool_.reset()` here ?

Reply:
Certainly!